qldbdriver/result.go

```diff
@@ -83,30 +84,27 @@ func (result *result) getNextPage() error {
 
 func (result *result) updateMetrics(fetchPageResult *qldbsession.FetchPageResult) {
 	if fetchPageResult.ConsumedIOs != nil {
-		*result.metrics.ioUsage.readIOs += *fetchPageResult.ConsumedIOs.ReadIOs
-		*result.metrics.ioUsage.writeIOs += *fetchPageResult.ConsumedIOs.WriteIOs
+		*result.ioUsage.readIOs += *fetchPageResult.ConsumedIOs.ReadIOs
+		*result.ioUsage.writeIOs += *fetchPageResult.ConsumedIOs.WriteIOs
 	}
 
 	if fetchPageResult.TimingInformation != nil {
-		*result.metrics.timingInformation.processingTimeMilliseconds += *fetchPageResult.TimingInformation.ProcessingTimeMilliseconds
+		*result.timingInfo.processingTimeMilliseconds += *fetchPageResult.TimingInformation.ProcessingTimeMilliseconds
 	}
 }
 
 // GetConsumedIOs returns the statement statistics for the current number of read IO requests that were consumed. The statistics are stateful.
 func (result *result) GetConsumedIOs() *IOUsage {
-	readIOs := *result.metrics.ioUsage.readIOs
-	writeIOs := *result.metrics.ioUsage.writeIOs
```

Comment:
I believe this is done to prevent the pointer of the actual data type from being overwritten if the user reassigned what they got back.

Please revert this part for this and ProcessingTime.

Reply:
Done.